Could not supply a percent symbol for webcontrol param update

There is no way to put a percent symbol as param value, for example, `text_left`:

```
$ curl http://localhost:8080/1/config/set?text_left=%
<!DOCTYPE html>
<html>
<body>
<p>Bad Request</p>
<p>The server did not understand your request.</p>
</body>
</html>
```
https://sourceforge.net/p/motion/mailman/message/37127567/

I have checked the PR, the issue seems resolved

Before the fix:

```
$ curl http://localhost:8080/%a
<!DOCTYPE html>
<html>
<body>
<p>Bad Request</p>
<p>The server did not understand your request.</p>
</body>
</html>


[0:cn0] [ERR] [STR] webu_url_decode: Error decoding url
free(): invalid pointer
Aborted (core dumped)


$ curl http://localhost:8081/%a
<!DOCTYPE html>
<html>
<body>
<p>Bad Request</p>
<p>The server did not understand your request.</p>
</body>
</html>


[0:cn0] [ERR] [STR] webu_url_decode: Error decoding url
free(): invalid pointer
Aborted (core dumped)
```

After the fix:
```
$ curl http://localhost:8080/%a
<!DOCTYPE html>
<html>
<body>
<p>Bad Request</p>
<p>The server did not understand your request.</p>
</body>
</html>

[0:wu0] [INF] [ALL] webu_clientip: Connection from: 127.0.0.1
[0:wu0] [INF] [STR] webu_html_main: Invalid action requested: >%a< >< ><


$ curl http://localhost:8081/%a
<html><head><title>Access denied</title></head><body>Access denied</body></html>

[0:st0] [INF] [ALL] webu_clientip: Connection from: 127.0.0.1
```

Ubuntu 14 build break:
```
src/webu.c:414: undefined reference to `MHD_http_unescape'
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -34,11 +34,6 @@ matrix:
     language: c
     compiler: gcc
   - os: linux
-    env:  DOCKER_IMAGE=debian:jessie
-    services:  docker
-    language: c
-    compiler: gcc
-  - os: linux
     env:  DOCKER_IMAGE=debian:stretch
     services:  docker
     language: c
@@ -91,14 +86,6 @@ before_script:
       docker exec $(docker ps -aq) /bin/bash -c 'apt-get install -y build-essential libjpeg62-turbo-dev libzip-dev autoconf automake autopoint pkgconf libtool git';
       docker exec $(docker ps -aq) /bin/bash -c 'apt-get install -y libavformat-dev libavcodec-dev libavutil-dev libswscale-dev libavdevice-dev';
       docker exec $(docker ps -aq) /bin/bash -c 'apt-get install -y libsqlite3-dev libpq-dev libmariadbclient-dev libwebp-dev libmicrohttpd-dev gettext';
-      docker exec $(docker ps -aq) /bin/bash -c 'autoreconf -fiv';
-    elif [ "$DOCKER_IMAGE" = "debian:jessie" ]; then
-      docker exec $(docker ps -aq) /bin/bash -c 'ln -fs /usr/share/zoneinfo/UTC /etc/localtime';
-      docker exec $(docker ps -aq) /bin/bash -c 'export DEBIAN_FRONTEND=noninteractive';
-      docker exec $(docker ps -aq) /bin/bash -c 'apt-get -qq update';
-      docker exec $(docker ps -aq) /bin/bash -c 'apt-get install -y build-essential libjpeg62-turbo-dev libzip-dev autoconf automake autopoint pkgconf libtool git';
-      docker exec $(docker ps -aq) /bin/bash -c 'apt-get install -y libavformat-dev libavcodec-dev libavutil-dev libswscale-dev libavdevice-dev';
-      docker exec $(docker ps -aq) /bin/bash -c 'apt-get install -y libsqlite3-dev libpq-dev libmysqlclient-dev libwebp-dev libmicrohttpd-dev gettext';
       docker exec $(docker ps -aq) /bin/bash -c 'autoreconf -fiv';
     elif [ "$DOCKER_IMAGE" = "ubuntu:16.04" ]; then
       docker exec $(docker ps -aq) /bin/bash -c 'ln -fs /usr/share/zoneinfo/UTC /etc/localtime';

--- a/src/webu.c
+++ b/src/webu.c
@@ -4,12 +4,6 @@
  *    Webcontrol and Streams for motion.
  *
  *    This software is distributed under the GNU Public License Version 2
- *    See also the file 'COPYING'.
- *
- *    Portions of code from Angel Carpintero (motiondevelop@gmail.com)
- *    from webhttpd.c Copyright 2004-2005
- *
- *    Majority of module written by MrDave.
  *
  *    Function naming scheme:
  *      webu*      - All functions in this module have this prefix.
@@ -231,66 +225,6 @@ void webu_write(struct webui_ctx *webui, const char *buf) {
     return;
 }
 
-static int webu_url_decode(char *urlencoded, size_t length) {
-    /* We are sent a URI encoded string and this decodes it to characters
-     * If the sent URL that isn't valid, then we clear out the URL
-     * so it is not processed in further functions.  The "answer" functions
-     * look for empty urls and answer with bad request
-     */
-    char *data = urlencoded;
-    char *urldecoded = urlencoded;
-    int scan_rslt;
-    size_t origlen;
-
-    origlen = length;
-
-    if (urlencoded[0] != '/'){
-        MOTION_LOG(ERR, TYPE_STREAM, NO_ERRNO, _("Invalid url: %s"),urlencoded);
-        memset(urlencoded,'\0',origlen);
-        return -1;
-    }
-
-    while (length > 0) {
-        if (*data == '%') {
-            char c[3];
-            int i;
-            data++;
-            length--;
-            c[0] = *data++;
-            length--;
-            c[1] = *data;
-            c[2] = 0;
-
-            scan_rslt = sscanf(c, "%x", &i);
-            if (scan_rslt < 1){
-                MOTION_LOG(ERR, TYPE_STREAM, NO_ERRNO,_("Error decoding url"));
-                memset(urlencoded,'\0',origlen);
-                return -1;
-            }
-
-            if (i < 128) {
-                *urldecoded++ = (char)i;
-            } else {
-                *urldecoded++ = '%';
-                *urldecoded++ = c[0];
-                *urldecoded++ = c[1];
-            }
-
-        } else if (*data == '<' || *data == '+' || *data == '>') {
-            *urldecoded++ = ' ';
-        } else {
-            *urldecoded++ = *data;
-        }
-
-        data++;
-        length--;
-    }
-    *urldecoded = '\0';
-
-    return 0;
-
-}
-
 static void webu_parms_edit(struct webui_ctx *webui) {
 
     /* Determine the thread number provided.
@@ -477,8 +411,7 @@ static int webu_parseurl(struct webui_ctx *webui) {
 
     if (strlen(webui->url) == 0) return -1;
 
-    retcd = webu_url_decode(webui->url, strlen(webui->url));
-    if (retcd != 0) return retcd;
+    MHD_http_unescape(webui->url);
 
     MOTION_LOG(DBG, TYPE_STREAM, NO_ERRNO, _("Decoded url: %s"),webui->url);
 

